flight/Modules/VtolPathFollower/vtol_follower_fsm.c

```diff
@@ -293,36 +373,18 @@ static void vtol_fsm_fsm_init(const struct vtol_fsm_transition *goal)
  * This method will first update the current state @ref curr_state based on the
  * current state and the active @ref current_goal. When it enters a new state it
  * calls the appropriate entry_fn if it exists.
+ *
+ * This differs from vtol_fsm_process_event in that it handles auto transitions
+ * afterwards.
  */
 static void vtol_fsm_inject_event(enum vtol_fsm_event event)
 {
-	// No need for mutexes here since this is called in a single threaded manner
-
-	/*
-	 * The STATE_UNCHANGED indicates to ignore these events
-	 */
-	if (current_goal[curr_state].next_state[event] == FSM_STATE_UNCHANGED)
```

Comment:
so a side effect of this change: an event that does not make a change will still process auto events. however, since any auto event for a state should have already happened that doesn't seem to matter.